internal/datacoord/compaction.go

```diff
@@ -78,6 +78,76 @@ type compactionInfo struct {
 	mergeInfos   map[int64]*milvuspb.CompactionMergeInfo
 }
 
+type NodeAssigner interface {
+	assign(t CompactionTask) bool
+}
+
+type SlotBasedNodeAssigner struct {
+	cluster Cluster
+
+	slots map[int64]int64
+}
+
+var _ NodeAssigner = (*SlotBasedNodeAssigner)(nil)
+
+func newSlotBasedNodeAssigner(cluster Cluster) *SlotBasedNodeAssigner {
+	return &SlotBasedNodeAssigner{
+		cluster: cluster,
+	}
+}
+
+func (sna *SlotBasedNodeAssigner) assign(t CompactionTask) bool {
+	if sna.slots == nil {
+		sna.slots = sna.cluster.QuerySlots()
+	}
+
+	log := log.Ctx(context.TODO())
+	nodeID, useSlot := sna.pickAnyNode(t)
+	if nodeID == NullNodeID {
+		log.Info("cannot find datanode for compaction task",
+			zap.Int64("planID", t.GetTaskProto().GetPlanID()),
+			zap.String("type", t.GetTaskProto().GetType().String()),
+			zap.String("vchannel", t.GetTaskProto().GetChannel()))
```

Comment:
Even though It's a Info level log, many people still worried about this message, maybe we could log about `requiredSlots` and `AvaialbleSlots` too. For example:

`failed to pick nodes for compaction task, insufficient slots available. Require xxx, Available xxx`